src/main/kotlin/com/dobby/backend/application/usecase/experiment/UpdateExperimentPostUseCase.kt

```diff
@@ -143,8 +146,18 @@ class UpdateExperimentPostUseCase (
         if(existingPost.member.id != input.memberId) throw PermissionDeniedException
     }
 
-    private fun validateNotExpired(existingPost: ExperimentPost){
-        if (!existingPost.recruitStatus) throw ExperimentPostUpdateDateException
+    private fun validateModificationPolicy(existingPost: ExperimentPost, input: Input){
+        if(!existingPost.recruitStatus){
+            if(input.startDate != null) {
+                throw ExperimentPostUpdateDateException
+            }
+            if(input.endDate != null) {
+                throw ExperimentPostUpdateDateException
+            }
+            if(input.recruitStatus != null) {
+                throw ExperimentPostRecruitStatusException
+            }
+        }
```

Comment:
```suggestion
    private fun validateModificationPolicy(existingPost: ExperimentPost, input: Input) {
        if (!existingPost.recruitStatus) {
            if (input.startDate != null || input.endDate != null) {
                throw ExperimentPostUpdateDateException
            }
            if (input.recruitStatus != null) {
                throw ExperimentPostRecruitStatusException
            }
        }
    }
```

각 경우에 대해 세분화된 예외를 던져주고 있네요. 👍 여기서 `startDate`와 `endDate` 둘중 하나만 null이 아니면 ExperimentPostUpdateDateException 던지도록 개선할 수 있어보여요!